FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/abstractions/platform/freertos/transport_interface_freertos.c

```diff
@@ -91,6 +77,27 @@ BaseType_t Transport_FreeRTOS_Connect( NetworkContext_t * pNetworkContext,
     if( socketStatus == 0 )
     {
         socketStatus = FreeRTOS_connect( tcpSocket, &serverAddress, sizeof( serverAddress ) );
+
+        if( socketStatus != 0 )
+        {
+            LogError( ( "Failed to connect to %s. FreeRTOS_connect returned with status=%d.",
+                        pHostName,
+                        socketStatus ) );
+        }
+    }
+
+    if( socketStatus == 0 )
+    {
+        socketStatus = FreeRTOS_setsockopt( tcpSocket,
```

Comment:
I think we should  add send timeout as well.

Reply:
Yea agree. I think that will be a good change to be in a separate PR.